enricher/api/src/main/java/io/fabric8/maven/enricher/api/BaseEnricher.java

```diff
@@ -133,8 +134,30 @@ protected void addInitContainer(PodTemplateSpecBuilder builder, JSONObject initC
         String initContainerAnnotation = builder.buildMetadata().getAnnotations().get(INIT_CONTAINER_ANNOTATION);
         JSONArray initContainers = Strings.isNullOrBlank(initContainerAnnotation) ? new JSONArray()
                 : new JSONArray(initContainerAnnotation);
-        initContainers.put(initContainer);
 
+        // lets avoid duplicates being added
+        boolean duplicate = false;
+        int length = initContainers.length();
+        for (int i = 0; i < length; i++) {
+            JSONObject obj = initContainers.getJSONObject(i);
+            if (duplicateInitcontainer(obj, initContainer)) {
+                duplicate = true;
+            }
+
+        }
+        if (!duplicate) {
+            initContainers.put(initContainer);
+        }
         builder.editMetadata().addToAnnotations(INIT_CONTAINER_ANNOTATION, initContainers.toString()).endMetadata();
     }
+
+    private boolean duplicateInitcontainer(JSONObject a, JSONObject b) {
+        Object name1 = a.get("name");
+        Object name2 = b.get("name");
+        if (Objects.equal(name1, name2)) {
+            log.warn("Found duplicate init containers with names " + name1 + " so ignoring " + b);
```

Comment:
@jimmidyson so a trivial bit of code in a single commit to avoid duplicates (which we probably will always need as duplicates totally fail the Deployment) is gonna take @rhuss longer time to deal with than manually testing fabric8-platform works with all fmp commits going forward? That sounds totally sound logic there :).  

BTW the reviewed refactorings of fmp which introduced this issue also wasted lots of my time too, which was nothing to do with me, I was just trying to get fabric8-platform released. Yes code reviews can help. But bugs slow us all down too. 

Anyway am hoping we can get automated CI soon though which should save us all time going forward